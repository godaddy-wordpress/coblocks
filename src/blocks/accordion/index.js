/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
