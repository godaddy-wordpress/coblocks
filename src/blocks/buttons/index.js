/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
