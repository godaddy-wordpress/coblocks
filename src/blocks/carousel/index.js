/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
