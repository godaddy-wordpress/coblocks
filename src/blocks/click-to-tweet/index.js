/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import { Icon } from '@wordpress/components';
import { TwitterIcon as icon } from '@godaddy-wordpress/coblocks-icons';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	deprecated,
	/* translators: block description */
	description: __( 'Add a quote for readers to tweet via Twitter.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			content: __( 'The easiest way to promote and advertise your blog, website, and business on Twitter.', 'coblocks' ),
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'share', 'coblocks' ),
		/* translators: block keyword */
		__( 'twitter', 'coblocks' ),
	],
	save,
	/* translators: block name */
	title: __( 'Click to Tweet', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
