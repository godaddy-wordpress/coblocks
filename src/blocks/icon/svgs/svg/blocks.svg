<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>