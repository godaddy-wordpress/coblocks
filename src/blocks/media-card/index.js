/**
 * Internal dependencies
 */
import { BackgroundAttributes } from '../../components/background';
import DimensionsAttributes from '../../components/dimensions-control/attributes';
import edit from './edit';
import { MediaCardIcon as icon } from '@godaddy-wordpress/coblocks-icons';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...BackgroundAttributes,
	...DimensionsAttributes,
	...metadata.attributes,
};

const settings = {
	/* translators: block name */
	title: __( 'Media Card', 'coblocks' ),
	/* translators: block description */
	description: __( 'Add an image or video with an offset card side-by-side.', 'coblocks' ),
	icon: <Icon icon={ icon } />,
	supports: {
		align: [ 'wide', 'full' ],
		stackedOnMobile: true,
		coBlocksSpacing: true,
	},
	example: {
		attributes: {
			align: 'wide',
			mediaType: 'image',
			mediaUrl: 'https://s.w.org/images/core/5.3/MtBlanc1.jpg',
			mediaWidth: 45,
		},
	},
	attributes,
	transforms,
	edit,
	save,
};

export { name, category, metadata, settings };
