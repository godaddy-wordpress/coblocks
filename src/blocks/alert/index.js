/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
