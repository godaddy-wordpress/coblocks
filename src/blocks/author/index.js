/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
