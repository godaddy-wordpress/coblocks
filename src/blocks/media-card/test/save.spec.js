/**
 * External dependencies
 */
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;

const DEFAULT_IMAGE_ATTRIBUTES = {
	backgroundImg: '150x150.png',
	backgroundOverlay: 50,
	backgroundRepeat: 'no-repeat',
	backgroundSize: 'contain',
	backgroundType: 'image',
	focalPoint: { x: 0.6352941176470588, y: 0.3729411405675552 },
	hasImgShadow: true,
	mediaPosition: 'right',
	mediaType: 'image',
	mediaUrl: '150x150.png',
};

describe( 'coblocks/media-card', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );
	} );

	it( 'should render', () => {
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with appropriate id class', () => {
		block.attributes.coblocks = { id: '1234' };
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'coblocks-media-card-1234' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with grid template column classes', () => {
		[ 'left', 'right' ].forEach( ( position ) => {
			block.attributes.mediaPosition = position;
			block.attributes.mediaWidth = '100';
			const serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( position === 'right' ? 'auto 100%' : '100% auto' );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should render with align attribute', () => {
		const alignOptions = [ 'full', 'wide' ];
		alignOptions.forEach( ( alignOption ) => {
			block.attributes.align = alignOption;
			block.attributes.maxWidth = '100px';
			const serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `align${ alignOption }` );
			expect( serializedBlock ).toContain( 'max-width:100px' );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should render with className attribute', () => {
		block.attributes.className = 'my-custom-class';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"className":"my-custom-class"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with background image attributes', () => {
		block.attributes = DEFAULT_IMAGE_ATTRIBUTES;
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'src="150x150.png"' );
		expect( serializedBlock ).toContain( 'background-image:url(150x150.png)' );
		expect( serializedBlock ).toContain( 'background-position:63.52941176470588% 37.29411405675552%' );
		expect( serializedBlock ).toContain( 'has-background-image' );
		expect( serializedBlock ).toContain( 'has-background-overlay' );
		expect( serializedBlock ).toContain( 'is-style-right' );
		expect( serializedBlock ).toContain( 'bg-no-repeat' );
		expect( serializedBlock ).toContain( 'has-shadow' );
		expect( serializedBlock ).toContain( 'bg-contain' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with background image attributes with media id', () => {
		block.attributes = DEFAULT_IMAGE_ATTRIBUTES;
		block.attributes.mediaId = '1234';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'wp-image-1234' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with background video attributes', () => {
		block.attributes.backgroundImg = '150x150.mp4';
		block.attributes.backgroundType = 'video';
		block.attributes.mediaUrl = '150x150.mp4';
		block.attributes.mediaType = 'video';
		block.attributes.mediaPosition = 'left';
		block.attributes.hasCardShadow = true;
		block.attributes.videoLoop = true;
		block.attributes.videoMuted = true;
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'src="150x150.mp4"' );
		expect( serializedBlock ).toContain( 'has-background-video' );
		expect( serializedBlock ).toContain( 'coblocks-video-bg' );
		expect( serializedBlock ).toContain( 'is-style-left' );
		expect( serializedBlock ).toContain( 'has-shadow' );
		expect( serializedBlock ).toContain( 'playsinline' );
		expect( serializedBlock ).toContain( 'autoplay' );
		expect( serializedBlock ).toContain( 'muted' );
		expect( serializedBlock ).toContain( 'loop' );
		expect( serializedBlock ).toContain( 'src="150x150.mp4"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render padding classes', () => {
		[ 'none', 'small', 'medium', 'large', 'huge' ].forEach( ( paddingSize ) => {
			block.attributes.paddingSize = paddingSize;
			const serializedBlock = serialize( block );
			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( 'has-' + paddingSize + '-padding' );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );
} );
