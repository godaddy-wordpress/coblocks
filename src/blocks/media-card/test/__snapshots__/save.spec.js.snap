// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/media-card should render 1`] = `
"<!-- wp:coblocks/media-card -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render padding classes 1`] = `
"<!-- wp:coblocks/media-card {\\"paddingSize\\":\\"none\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-padding has-none-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render padding classes 2`] = `
"<!-- wp:coblocks/media-card {\\"paddingSize\\":\\"small\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-padding has-small-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render padding classes 3`] = `
"<!-- wp:coblocks/media-card {\\"paddingSize\\":\\"medium\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-padding has-medium-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render padding classes 4`] = `
"<!-- wp:coblocks/media-card {\\"paddingSize\\":\\"large\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-padding has-large-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render padding classes 5`] = `
"<!-- wp:coblocks/media-card {\\"paddingSize\\":\\"huge\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-padding has-huge-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with align attribute 1`] = `
"<!-- wp:coblocks/media-card {\\"align\\":\\"full\\",\\"maxWidth\\":\\"100px\\"} -->
<div class=\\"wp-block-coblocks-media-card alignfull is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"max-width:100px\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with align attribute 2`] = `
"<!-- wp:coblocks/media-card {\\"maxWidth\\":\\"100px\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"max-width:100px\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with appropriate id class 1`] = `
"<!-- wp:coblocks/media-card -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media coblocks-media-card-1234\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with background image attributes 1`] = `
"<!-- wp:coblocks/media-card {\\"backgroundImg\\":\\"150x150.png\\",\\"backgroundSize\\":\\"contain\\",\\"backgroundOverlay\\":50,\\"focalPoint\\":{\\"x\\":0.6352941176470588,\\"y\\":0.3729411405675552},\\"mediaPosition\\":\\"right\\",\\"mediaType\\":\\"image\\",\\"hasImgShadow\\":true} -->
<div class=\\"wp-block-coblocks-media-card is-style-right\\"><div class=\\"wp-block-coblocks-media-card__inner has-background-overlay bg-contain has-background-image bg-no-repeat\\" style=\\"background-image:url(150x150.png);background-position:63.52941176470588% 37.29411405675552%\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"grid-template-columns:auto undefined%\\"><figure class=\\"wp-block-coblocks-media-card__media has-shadow\\"><img src=\\"150x150.png\\"/></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with background image attributes with media id 1`] = `
"<!-- wp:coblocks/media-card {\\"backgroundImg\\":\\"150x150.png\\",\\"backgroundSize\\":\\"contain\\",\\"backgroundOverlay\\":50,\\"focalPoint\\":{\\"x\\":0.6352941176470588,\\"y\\":0.3729411405675552},\\"mediaPosition\\":\\"right\\",\\"mediaId\\":\\"1234\\",\\"mediaType\\":\\"image\\",\\"hasImgShadow\\":true} -->
<div class=\\"wp-block-coblocks-media-card is-style-right\\"><div class=\\"wp-block-coblocks-media-card__inner has-background-overlay bg-contain has-background-image bg-no-repeat\\" style=\\"background-image:url(150x150.png);background-position:63.52941176470588% 37.29411405675552%\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"grid-template-columns:auto undefined%\\"><figure class=\\"wp-block-coblocks-media-card__media has-shadow\\"><img src=\\"150x150.png\\" class=\\"wp-image-1234\\"/></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with background video attributes 1`] = `
"<!-- wp:coblocks/media-card {\\"backgroundType\\":\\"video\\",\\"backgroundImg\\":\\"150x150.mp4\\",\\"mediaType\\":\\"video\\",\\"hasCardShadow\\":true} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left\\"><div class=\\"wp-block-coblocks-media-card__inner has-background-video has-no-padding\\"><div class=\\"coblocks-video-bg position-absolute overflow-hidden w-full h-full pin-t pin-r pin-b pin-l\\"><video class=\\"w-full h-full bg-center-center object-cover object-position\\" playsinline autoplay muted loop src=\\"150x150.mp4\\"></video></div><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"><video controls src=\\"150x150.mp4\\"></video></figure><div class=\\"wp-block-coblocks-media-card__content has-shadow\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with className attribute 1`] = `
"<!-- wp:coblocks/media-card {\\"className\\":\\"my-custom-class\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media my-custom-class\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with grid template column classes 1`] = `
"<!-- wp:coblocks/media-card {\\"mediaWidth\\":\\"100\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-left has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"grid-template-columns:100% auto\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;

exports[`coblocks/media-card should render with grid template column classes 2`] = `
"<!-- wp:coblocks/media-card {\\"mediaPosition\\":\\"right\\",\\"mediaWidth\\":\\"100\\"} -->
<div class=\\"wp-block-coblocks-media-card alignwide is-style-right has-no-media\\"><div class=\\"wp-block-coblocks-media-card__inner has-no-padding\\"><div class=\\"wp-block-coblocks-media-card__wrapper\\" style=\\"grid-template-columns:auto 100%\\"><figure class=\\"wp-block-coblocks-media-card__media\\"></figure><div class=\\"wp-block-coblocks-media-card__content\\"></div></div></div></div>
<!-- /wp:coblocks/media-card -->"
`;
